doc_internal/support-tiers.md

```diff
@@ -0,0 +1,119 @@
+# "Support" Tierlist
```

Comment:
in future it will be nice to have an externally-facing document of what we are looking for in a terminal, i.e.
- required things like
  - termcaps we need for core functionality
  - the ability to parse arbitrary CSI and OSC commands
- nice-to have features like OSC 133 etc.

then any terminal developer that wants to fully support fish can simply tick
all the boxes in that list and be done, no need to read source code.